caluma/caluma_core/health_checks.py

```diff
@@ -0,0 +1,213 @@
+        # Cleanup
+        out.close()
+
+
+@check
+def _check_minio():
+    """
+    Check minio connection and operations.
+    """
+
+    minio_client = storage_clients.client
+    response = None
+
+    # Check minio operations
+    try:
+        object_name = "my-object"
+        new_object_name = "my-object-new"
+
+        # create bucket (storage client doesn't create bucket before)
+        minio_client.stat_object(object_name)
+
+        # Create object
+        data = b"hello"
+        minio_client.client.put_object(
+            minio_client.bucket,
+            object_name,
+            BytesIO(data),
+            5,
+        )
+
+        # Object and information retrieval
+        stat = minio_client.stat_object(object_name)
+        assert stat.object_name == "my-object"
+
+        response = minio_client.client.get_object(minio_client.bucket, object_name)
+        assert response.data == data
+
+        # Move object
+        minio_client.move_object(object_name, new_object_name)
+        assert not minio_client.stat_object(object_name)
+
+        # Remove object
+        minio_client.remove_object(new_object_name)
+        assert not minio_client.stat_object(new_object_name)
+
+        return {"ok": True}
+
+    finally:
+        # Cleanup
+        if response:
+            response.close()
+            response.release_conn()
+
+        minio_client.remove_object(new_object_name)
+
+
+@check
+def _check_models(database):
+    """
+    Check model instantion and operations on database.
+    """
+
+    log = None
+    try:
+        # Instantiate model
+        log = AccessLog.objects.create(status_code=22)
+        assert log in list(
+            AccessLog.objects.all()
+        ), "Failed to save model instance to database."
+
+        # Retrieve object
+        retrieved_log = AccessLog.objects.get(pk=log.pk)
+        assert log == retrieved_log, "Unexpected object retrieved from database."
+        assert (
+            retrieved_log.status_code == log.status_code
+        ), "Unexpected object data retrieved from database."
+
+        # Update object
+        log.status_code = 42
+        log.save()
+        retrieved_log = AccessLog.objects.get(pk=log.pk)
+        assert log == retrieved_log, "Unexpected object retrieved from database."
+        assert (
+            retrieved_log.status_code == log.status_code
+        ), "Unexpected object data retrieved from database."
+
+        # Remove object
+        log.delete()
+        log = None
+        assert (
+            log not in AccessLog.objects.all()
+        ), "Failed to delete model instance from database."
+
+        return {database: {"ok": True}}
+
+    finally:
+        # Cleanup
+        if log:  # pragma: no cover
+            log.delete()
+
+
+def check_minio():
+    """
+    Check minio connection and operations.
+    """
+
+    # Check if minio is available
+    if not storage_clients.client:
+        return {"minio (not implemented)": {"ok": True}}
```

Comment:
Maybe change the string to "minio (not configured)" so people don't assume the minio check isn't implemented